src/Akka.Persistence.Sql/Query/SqlReadJournal.cs

```diff
@@ -219,15 +219,15 @@ private Source<EventEnvelope, NotUsed> EventsByPersistenceIdSource(
                             timestamp: r.representation.Timestamp, 
                             tags: r.tags));
 
-        private Source<EventEnvelope, NotUsed> CurrentJournalEvents(long offset, long max, MaxOrderingId latestOrdering)
+        private Source<EventEnvelope[], NotUsed> CurrentJournalEvents(long offset, long max, MaxOrderingId latestOrdering)
```

Comment:
Instead of using `SelectMany` and return a list of `EventEnvelope`s, we use `Select` and return a list of `EventEnvelope[]` arrays instead. This list count will be consistently equals to the number of rows we fetched from the database.